### PR TITLE
Updated Autocomplete Tests to Share Index

### LINKED ISSUE #11145: Reduce Index Thrashing in Search Live Tests
PR #11040 updated the Search live tests to use an index per test (when needed) instead of the previous logic of using a Search service per test. This was a basic conversion which left opportunities for tests to share a static instance when they are only doing read operations. Reducing the amount of index creation and deletion is important as Search services are limited on the number of operations per minute, right now that is 12 create and 12 delete operations. Using static indexes will make test runs more robust as they are less likely to be throttled and they will become quicker as they are paused for document indexing when creating the index.

Known candidates for this conversion are the following (this list will be updated as more are discovered).

- AutocompleteSyncTests

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #11145 

This PR updates the autocomplete tests to share a single index instance as they are read only tests and won't modify the data in the index.

### CODE REVIEW

File: sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/AutocompleteSyncTests.java [hunk 1]
```diff
@@ -23,20 +41,59 @@
...
+        try {
+            SearchIndex index = new ObjectMapper().readValue(indexData, SearchIndex.class);
+
+            Field searchIndexName = index.getClass().getDeclaredField("name");
+            AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
+                searchIndexName.setAccessible(true);
+                return null;
+            });
+
+            searchIndexName.set(index, INDEX_NAME);
```
Comment (sima-zhu) [inline sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/AutocompleteSyncTests.java:72]: Why can't we change the name directly in hotel json file but to modify here after serialization?
There are two other test classes which have the similar structure, SuggestSyncTests and SearchSyncTest which can do the similar stuff.

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/AutocompleteSyncTests.java]: Updated Search and Suggest test which are able to share an instance. The name is changed using reflection as many tests use this file so it won't be safe to make it a global name.

**Pull request merged**